navpi_diff_drive_controller:
  pid: {
    yaw: {
      p: {
        type: double,
        default_value: 0.0,
        description: "Proportional gain of the PID controller"
      },
      i: {
        type: double,
        default_value: 0.0,
        description: "Integral gain of the PID controller"
      },
      d: {
        type: double,
        default_value: 0.0,
        description: "Derivative gain of the PID controller"
      },
      i_max: {
        type: double,
        default_value: 0.0,
        description: "I max of the PID controller"
      },
      i_min: {
        type: double,
        default_value: 0.0,
        description: "I min of the PID controller"
      },
      anti_windup: {
        type: bool,
        default_value: false,
        description: "Anti windup of the PID controller"
      }
    }
  }
  bounds: {
    linear: {
      x: { type: double, default_value: 0.5, description: "", validation: {bounds<>: [0.0, 10.0]} },
      y: { type: double, default_value: 0.5, description: "", validation: {bounds<>: [0.0, 10.0]} },
      z: { type: double, default_value: 0.5, description: "", validation: {bounds<>: [0.0, 10.0]} },
    },
    angular: {
      x: { type: double, default_value: 1.0, description: "", validation: {bounds<>: [0.0, 10.0]} },
      y: { type: double, default_value: 1.0, description: "", validation: {bounds<>: [0.0, 10.0]} },
      z: { type: double, default_value: 1.0, description: "", validation: {bounds<>: [0.0, 10.0]} },
    }
  }
  rotate_in_movement_direction: {
    type: bool,
    default_value: true,
    description: "if set, tries to rotate the robot in the direction of movement",
  }
  only_forward: {
    type: bool,
    default_value: true,
    description: "if set, robot is only allowed to move forward",
  }
